address.js

```diff
@@ -203,6 +210,7 @@ define(function(require) {
         , body : body
         , context : context
         , origin: origin
+        , timeout: timeout
```

Comment:
I don't particularly like how we keep polluting the request object with additional data just to use it as a transport mechanism, it smells and it smells bad. That said, the precedence was set with `context` and `origin`, and doing something different with `timeout` is probably overkill for this PR, but we should consider rewriting this altogether at some point.

Reply:
Agreed.